src/poller_base.cpp

```diff
@@ -81,12 +82,13 @@ uint64_t zmq::poller_base_t::execute_timers ()
     //  Get the current time.
     const uint64_t current = _clock.now_ms ();
 
-    //   Execute the timers that are already due.
-    const timers_t::iterator begin = _timers.begin ();
-    const timers_t::iterator end = _timers.end ();
+    //  Execute the timers that are already due.
     uint64_t res = 0;
-    timers_t::iterator it = begin;
-    for (; it != end; ++it) {
+    timer_info_t timer_temp;
+    const timers_t::iterator end = _timers.end ();
```

Comment:
Strictly speaking, this iterator might also be invalidated when `_timers` is modified. (This might not be the case for the implementations used)

Reply:
`std::multimap` guarantees that only iterators of deleted elements are invalidated and others stay valid - [https://en.cppreference.com/w/cpp/container/multimap/erase](https://en.cppreference.com/w/cpp/container/multimap/erase).
I don't think that underlying container can be changed because the `execute_timers ()` code relies on `std::multimap` properties.
However, the code can be changed to "refresh" the `end ()` iterator on every iteration, there will be some overhead.

Reply:
Oh, sorry, then I mixed that up with other containers. Thanks for checking it again. Then I also think it should be kept this way.